src/main/webapp/app/entities/flair-bi/modal/modal-tabs/condition/filter-date-range-constraints.component.js

```diff
@@ -0,0 +1,238 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('flairbiApp')
+        .component('filterDateRangeConstraints', {
+            templateUrl: 'app/entities/flair-bi/modal/modal-tabs/condition/filter-date-range-constraints.component.html',
+            controller: FilterDateRangeConstraintsController,
+            controllerAs: 'vm',
+            bindings: {
+                dimension: '<',
+                reload: '<',
+                onDateChange: '&',
+                condition: '<',
+            }
+        });
+
+    FilterDateRangeConstraintsController.$inject = ['$scope'];
+
+    function FilterDateRangeConstraintsController($scope) {
+        var TAB_DAY = 0;
+        var TAB_RANGE = 1;
+        var TAB_DYNAMIC = 2;
+
+        var DYNAMIC_DATE_RANGE_CONFIG = [
+            {
```

Comment:
create common list which i can resued in sidebar and data constraints...it will be duplicate